nixos/modules/tasks/filesystems.nix

```diff
@@ -354,7 +354,15 @@ in
           options.
         '';
       }
-    ];
+    ] ++ lib.map (fs: {
+      assertion = fs.label != null -> fs.device == "/dev/disk/by-label/${fs.label}";
```

Comment:
I'm pretty sure you need to add an `escape` call here to match the one in the `mkDefault` definition.

Reply:
You're right, done.